concourse/tasks/update-ecs-service.yml

```diff
@@ -28,14 +28,30 @@ run:
       credential_source = Ec2InstanceMetadata
       EOF
 
+      current_task_definition_arn="$(aws ecs describe-services --services "$APPLICATION" --cluster govuk --region "$AWS_REGION" | jq -r '.services[0].taskDefinition')"
+      if [ -z "${current_task_definition_arn}"]; then
+        echo "failed to retrieve current task definition for ${APPLICATION}, exiting..."
+        exit 1
+      fi
+
+      new_task_definition_arn="$(cat "terraform-outputs/${APPLICATION}_task_definition_arn")"
+      if [ -z "${new_task_definition_arn}"]; then
+        echo "failed to retrieve new task definition for ${APPLICATION}, exiting..."
+        exit 1
+      fi
+
+      if [ "${current_task_definition_arn}" = "${new_task_definition_arn}" ]; then
+        echo "No need to update ${APPLICATION} service since its task definition was not updated"
+        exit 0
+      fi
```

Comment:
I don't think this bit is necessary, since if the task definition is the same, then `aws ecs update-service` is a no-op.

Reply:
Not what we experienced yesterday...

Reply:
For example, the ECS Service `content-store` is currently running task definition `content-store:61`.

```
aws ecs update-service --cluster govuk --service content-store --task-definition content-store:61
ecs wait services-stable --cluster govuk --services content-store # will exit immediately.
```

Reply:
If the service was not stable when `ecs update-service` was run, this might cause `ecs awit services-stable` to wait, even if the task definition had not changed.

Reply:
after further digging, I agree with you that it should be a no-op under normal circumstances if the task definition is the same. I think we got confused by the command printing the logs of the events which are older ones and not triggered by this change.

I think it is still could that we include the conditional above if you don't have an objection.

Reply:
If this additional complexity is indeed unnecessary then I have an objection to keeping it, unless there's a clear rationale for doing so.

Reply:
Ah, just saw your explanation in another thread below. So this basically just gets us slightly nicer-looking logs. I think I'm ok with either way then, but if you want to leave it in then please add a comment explaining what it's really for - because otherwise there's no way that somebody reading that shell script later would be able to know that that check/comparison logic isn't essential.

Reply:
I can do that.